lib/connections/service.go

```diff
@@ -268,15 +267,8 @@ next:
 			continue next
 		}
 
-		// If rate limiting is set, and based on the address we should
-		// limit the connection, then we wrap it in a limiter.
-
-		wr := io.Writer(c)
-		rd := io.Reader(c)
-		if s.shouldLimit(c.RemoteAddr()) {
-			wr = s.limiter.newWriteLimiter(c)
-			rd = s.limiter.newReadLimiter(c)
-		}
+		wr := s.limiter.newWriteLimiter(c, s.isLAN(c.RemoteAddr()))
+		rd := s.limiter.newReadLimiter(c, s.isLAN(c.RemoteAddr()))
```

Comment:
I'd store the return value in a bool, just in case that function does something more expensive in thefuture